### ISSUE TITLE
Feature: Support empty label column headers

### ISSUE THREAD

**Issue by pwang347** [adazzle/react-data-grid#3624]
## Use case 
Sometimes columns will not have any visible header text labels, e.g., for an index column. Currently, using `role=columnheader` will trigger accessibility failures such as https://dequeuniversity.com/rules/axe/4.5/empty-table-header

## Proposed solution 
Consider adding an option to change the role to `cell`.

**Comment by amanmahajan7** [comment]
Can we use `name: <div aria-label="Index Column" />`

**Comment by pwang347** [comment]
Hi @amanmahajan7, thanks for the reply! From the guidance in the link above it's not acceptable to have just an aria-label:
<img width="943" alt="image" src="https://github.com/user-attachments/assets/3c478019-113e-4bcb-b055-a7f2a0f1a9f6">

I believe changing the role is equivalent to changing from `th` to `td`

**Comment by amanmahajan7** [comment]
Are you using RDG? It looks like you have a `table`?

**Comment by pwang347** [comment]
Yes I am using RDG, the above is just from the accessibility page. [axe-core](https://github.com/dequelabs/axe-core) creates the above violation for RDG, and seems like it is because RDG is always assigning `role=columheader`

**Comment by pwang347** [comment]
Hi @amanmahajan7, please see the following branch: https://github.com/pwang347/react-data-grid/tree/pawang/axe-test

Repro steps:
1. Checkout the branch and run `npm i && npm run start`
2. Click on the "No rows" sample on the demo website
3. Run the following code in the dev tools console
```typescript
(await axe.run([document.querySelector("body")])).violations
```

I get the following result:
```json
[
    {
        "id": "empty-table-header",
        "impact": "minor",
        "tags": [
            "cat.name-role-value",
            "best-practice"
        ],
        "description": "Ensure table headers have discernible text",
        "help": "Table header text should not be empty",
        "helpUrl": "https://dequeuniversity.com/rules/axe/4.10/empty-table-header?application=axeAPI",
        "nodes": [
            {
                "any": [
                    {
                        "id": "has-visible-text",
                        "data": null,
                        "relatedNodes": [],
                        "impact": "minor",
                        "message": "Element does not have text that is visible to screen readers"
                    }
                ],
                "all": [],
                "none": [],
                "impact": "minor",
                "html": "<div role=\"columnheader\" aria-colindex=\"2\" aria-rowspan=\"1\" aria-selected=\"false\" tabindex=\"-1\" class=\"rdg-cell cell_cj343x0\" style=\"inset-block-start: 0px; grid-area: 1 / 2 / 2 / 3; padding-block-start: calc(0 * var(--rdg-header-row-height));\"></div>",
                "target": [
                    "div[aria-colindex=\"2\"]"
                ],
                "failureSummary": "Fix any of the following:\n  Element does not have text that is visible to screen readers"
            }
        ]
    }
]
```

Thanks!
